partition_version.cc

```diff
@@ -200,7 +201,10 @@ stop_iteration partition_snapshot::merge_partition_versions() {
         }
         while (auto prev = current->prev()) {
             region().allocator().invalidate_references();
-            if (current->partition().apply_monotonically(*schema(), std::move(prev->partition()), _tracker, is_preemptible::yes) == stop_iteration::no) {
+            mutation_application_stats app_stats;
+            const auto do_stop = current->partition().apply_monotonically(*schema(), std::move(prev->partition()),
+                    _tracker, app_stats, is_preemptible::yes);
+            if (do_stop == stop_iteration::no) {
                 return stop_iteration::no;
```

Comment:
Why the big code change?

We could (later) also collect these merging/cleaning statistics.

Reply:
Removed temporary `do_stop` variable.

Reply:
Re-added the variable, but it actually is needed now because of the addition-assignment.